### PR TITLE
6 packages from dialohq/ocaml-grpc at 0.2.0

### PR DESCRIPTION
**Pull request by quernd**
This pull-request concerns:
- `grpc.0.2.0`: A modular gRPC library
- `grpc-async.0.2.0`: An Async implementation of gRPC
- `grpc-bench.0.2.0`: grpc benchmarking
- `grpc-eio.0.2.0`: An Eio implementation of gRPC
- `grpc-examples.0.2.0`: Various grpc examples
- `grpc-lwt.0.2.0`: An Lwt implementation of gRPC

---
* Homepage: https://github.com/dialohq/ocaml-grpc
* Source repo: git+https://github.com/dialohq/ocaml-grpc.git
* Bug tracker: https://github.com/dialohq/ocaml-grpc/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

### CODE REVIEW

**Comment by quernd** [comment]
We split the `grpc-bench` and `grpc-examples` into separate packages. Does it make sense to have them in OPAM or will it clutter the index? My intuition is that developers who wish to use these packages are more likely to check out the repository instead of installing them.

**Comment by haochenx** [comment]
> We split the grpc-bench and grpc-examples into separate packages. Does it make sense to have them in OPAM or will it clutter the index? 

I believe that you're encouraged to publish them on OPAM if they provide value to other developers by making them easily installable via OPAM. Since the packages are all prefixed by `grpc-`, I am not particularly concerned about cluttering.

Potentially making it more demanding for the CI is probably the only concern here, but building them on the CI might give a better correctness guarantee. 

So unless `grpc-bench` or `grpc-examples` are adding too much stress for the CI, I'd like to say that it's your choice (but other opam-repo maintainers may have different opinions.)

**Review by haochenx — APPROVED** [review]
All looks good to me. The CI failures all seem to be irrelevant (`(failed: Library "h2-async" not found.`)

**Comment by quernd** [comment]
> All looks good to me. The CI failures all seem to be irrelevant (`(failed: Library "h2-async" not found.`)

It's an oversight on our part. The `dune` file for `grpc-async` shouldn't depend on `h2-async` like it does now:
https://github.com/dialohq/ocaml-grpc/blob/6eb6f6ddd5997b9de14526fb815b1d6e9b2bab00/lib/grpc-async/dune#L4
since `h2-async` is not listed as a dependency in `grpc-async.opam`.

It's meant to be used with `h2-async` but technically doesn't depend on it because the user is responsible for setting up the HTTP2 transport, and all our gRPC code uses the backend-agnostic `h2` only.

We need to clean that up, please don't merge yet.

**Comment by haochenx** [comment]
Oops. Sorry for the mistake. I somehow mistaken that as a native library not being available.

Thanks for the spot! @quernd

**Comment by quernd** [comment]
@haochenx Thanks, no worries! I worked around it by adding the dependency to the `.opam` file for now. We'll fix it properly in a patch release. The remaining CI failures are lint (we'll also deal with that in the patch release) and this one:

* distributions
   * oraclelinux-9-ocaml-5.1
        [grpc-examples.0.2.0 (failed: conf-protoc.4.4.0 failed to build)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/36251172ba01441aeeb55d21c62ab267d03935ef/variant/distributions,oraclelinux-9-ocaml-5.1,grpc-examples.0.2.0)

```[ERROR] The compilation of conf-protoc.4.4.0 failed at "protoc --version".```

File: packages/grpc-bench/grpc-bench.0.2.0/opam
Comment (haochenx) [inline packages/grpc-bench/grpc-bench.0.2.0/opam:3]: CI linter is giving the the following warning:
`warning 47: Synopsis should start with a capital and not end with a dot`

I don't think this is a blocker but fixing it will help us by reducing CI noises so we'd appreciate if you can consider about it.

**Review by haochenx — APPROVED** [review]
Now it's all good. The only real CI failure is `grpc-examples.0.2.0 (failed: conf-protoc.4.4.0 failed to build)` on `oraclelinux-9-ocaml-5.1` which is a missing depexts problem so not related to this PR.

The author may want to address the linter warning but I don't think it's a blocker at this point.

File: packages/grpc-bench/grpc-bench.0.2.0/opam
Comment (quernd) [reply packages/grpc-bench/grpc-bench.0.2.0/opam:3]: Absolutely, this is already in progress in https://github.com/dialohq/ocaml-grpc/pull/43/files#diff-d71eddca4e2da37e56aa8fc421b39f58c01b477013569e977aeb6af77d37f3b7 and will be fixed in the next release.

Comment (avsm) [reply packages/grpc-bench/grpc-bench.0.2.0/opam:3]:
```suggestion
synopsis: "Benchmarking package for gRPC"
```

We can just fix it in this PR as well, so that it doesnt look odd when rendered on the package website.

**Comment by mseri** [comment]
Thanks

**Comment by tmcgilchrist** [comment]
Thanks everyone!

**Pull request merged**
